src/parser.ts

```diff
@@ -69,7 +69,21 @@ class V1Validator {
    * This saves us a lot of code but might be a little bit slower.
    */
   validateNode(node: any) {
-    this.validateTypedInterface(node)
+    const currentPath = this.getLocation()
+
+    // allow defining properties/exports/imports/schemas named `type` or `format`
+    const skipPatterns = [
+      /^#\/components\/schemas\/[^/]+\/properties$/,
+      /^#\/components\/schemas$/,
+      /^#\/components$/,
+      /^#\/exports$/,
+      /^#\/imports$/
+    ]
+    
+    const shouldValidate = skipPatterns.none(pattern => pattern.test(currentPath))
+    if (shouldValidate) {
+      this.validateTypedInterface(node)
+    }
```

Comment:
Can't we offload some of this to the JSON schema validator? This PR description has some of my thinking here: https://github.com/dylibso/xtp-bindgen/pull/18

We should be able to assume that the documents that come into here are shaped appropriately and don't have illegal properties (missing or additional). I worry about littering the code with lots of extra checks when perhaps the JSON Schema validation is better suited for this kind of task.

Reply:
not that although this is an `any` object. It only is any for convenience. When we parse a document into a V1Schema we should have confidence that it is valid.

Reply:
Also to echo the document and comments here. I see validation in the parser module as checks that cannot be easily performed at the json schema level. E.g.: they require too much conditional logic.

Reply:
While these documents are valid, we need to skip this.validateTypedInterface for certain nodes - specifically those that are user-defined maps. The reason is that these maps can contain keys like type and format, which would trigger unwanted validation checks by this.validateTypedInterface.
Additionally, I've added /^#\/components$/ to the exclusion list as a precaution, in case we later introduce components with type or format fields in the schema.

So this code is not running validation, it's skipping them!